frontend/src/services/StorageService.ts

```diff
@@ -113,6 +113,20 @@ async function uploadDataset(
   };
 }
 
+async function uploadMetric(jsonFile: string): Promise<UploadDatasetResult> {
```

Comment:
An alternative more generic name for this function could be `uploadJson`, because there is nothing specific about metrics in this function other than simply uploading a JSON file.

Reply:
This function calls a function named uploadJson already.  Maybe we should discuss this module later, but I essentially left out the s3 key generation logic outside of the uploadJson function since it seemed like the behavior could change among different items that were needed to upload. 

For example: dataset - the keys were shared for the uploadJson and uploadFile. So I thought it made sense to make a function called "uploadDataset" that would just call uploadFile + uploadJson with the keys passed in. @miguelpdiaz8 is following the pattern that I started.

Should we all discuss later if we should refactor?

Reply:
Yes, as Andrew said we already have a function that specifically upload a json to S3. This new function uploadMetric encapsulate the key generation and then upload the json to S3 using the uploadJson function. We can discuss later.